src/app/plugins/animal.plugin.ts

```diff
@@ -0,0 +1,81 @@
+import Constants from '../../common/constants';
+import { Plugin } from '../../common/plugin';
+import { ChannelType, IContainer, IMessage, IHttpResponse} from '../../common/types';
+import { RichEmbed } from 'discord.js';
+
+export class AnimalPlugin extends Plugin {
+  public name: string = 'Animal Plugin';
+  public description: string = 'Delivers pictures of our furry friends!';
+  public usage: string = 'animal';
+  public pluginAlias = ['animal', "Animal"];
+  public permission: ChannelType = ChannelType.Public;
+  public pluginChannelName: string = Constants.Channels.Public.Pets;
+
+  private _API_URL: string = 'https://fetchit.dev/breeds/';
+  constructor(public container: IContainer) {
+    super();
+  }
+
+  public async execute(message: IMessage, args?: string[]) {
+    const parsedArgs = (args || []).map((str) => str.toLowerCase()).join(' ');
```

Comment:
get the parse function from the dog or cat API, will hide this mess here (line 20, 21)